src/mw/device/sort_archetype.cpp

```diff
@@ -982,6 +982,29 @@ void SortArchetypeNodeBase::sortSetup(int32_t)
     StateManager *state_mgr = mwGPU::getStateManager();
     int32_t num_columns = state_mgr->getArchetypeNumColumns(archetypeID);
 
+    // If this is not a world sort, we want to perform the sort always
+    bool world_sort = sortColumnIndex == 1 /* 1 is the WorldID column */;
+
+    if (!state_mgr->archetypeNeedsSort(archetypeID) && world_sort) {
+        numDynamicInvocations = 0;
+
+        for (int i = 0; i < numPasses; i++) {
+            taskgraph.getNodeData(onesweepNodes[i]).numDynamicInvocations = 0;
+        }
+        return;
+    }
+
+    if (world_sort) {
+        state_mgr->archetypeClearNeedsSort(archetypeID);
+    } else {
+        // If this isn't a world sort, this sort will scramble the entities
+        // across worlds and therefore we need to set the needs sort
+        // (which denotes whether the entity needs to be sorted by world)
+        // to true
+        state_mgr->archetypeSetNeedsSort(archetypeID);
+    }
+
+#if 0
```

Comment:
The sort is so critical to system correctness that let's not leave random commented stuff out in it. If the above code correctly replaces the commented out code, delete the commented out code.

Reply:
resolved

Reply:
This doesn't seem resolved? There's still commented out code

Reply:
sorry now it's removed